node/nodeTransactions.go

```diff
@@ -164,28 +141,31 @@ func (n *Node) unmarshalTransaction(txBytes []byte, txType transactionType) (*tr
 		if err != nil {
 			return nil, err
 		}
-		return n.prepareNormalTx(&tx)
+		status := n.computeTransactionStatus(&tx, false)
+		return n.prepareNormalTx(&tx, status)
 	case rewardTx:
 		var tx rewardTxData.RewardTx
 		err := n.internalMarshalizer.Unmarshal(&tx, txBytes)
 		if err != nil {
 			return nil, err
 		}
-		return n.prepareRewardTx(&tx)
+		status := n.computeTransactionStatus(&tx, false)
+		return n.prepareRewardTx(&tx, status)
 
 	case unsignedTx:
 		var tx smartContractResult.SmartContractResult
 		err := n.internalMarshalizer.Unmarshal(&tx, txBytes)
 		if err != nil {
 			return nil, err
 		}
-		return n.prepareUnsignedTx(&tx)
+		status := n.computeTransactionStatus(&tx, false)
+		return n.prepareUnsignedTx(&tx, status)
 	default:
 		return &transaction.ApiTransactionResult{Type: string(invalidTx)}, nil // this shouldn't happen
 	}
 }
 
-func (n *Node) prepareNormalTx(tx *transaction.Transaction) (*transaction.ApiTransactionResult, error) {
+func (n *Node) prepareNormalTx(tx *transaction.Transaction, status string) (*transaction.ApiTransactionResult, error) {
```

Comment:
We have the type `TransactionStatus` defined in `constants.go`. Can we use that one perhaps (also below)?

Reply:
done.